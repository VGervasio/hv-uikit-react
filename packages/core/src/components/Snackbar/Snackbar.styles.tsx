import styled from "@emotion/styled";
import Snackbar, { snackbarClasses } from "@mui/material/Snackbar";
import { theme } from "@hitachivantara/uikit-styles";

export const StyledSnackbar = styled(Snackbar)({
  [`&.${snackbarClasses.anchorOriginTopRight}`]: {
    top: theme.spacing(1),
    right: theme.spacing(1),
  },
  [`&.${snackbarClasses.anchorOriginTopLeft}`]: {
    top: theme.spacing(1),
    left: theme.spacing(1),
  },
  [`&.${snackbarClasses.anchorOriginTopCenter}`]: {
    top: theme.spacing(1),
  },
  [`&.${snackbarClasses.anchorOriginBottomCenter}`]: {
    bottom: theme.spacing(1),
  },
  [`&.${snackbarClasses.anchorOriginBottomLeft}`]: {
    bottom: theme.spacing(1),
    left: theme.spacing(1),
  },
  [`&.${snackbarClasses.anchorOriginBottomRight}`]: {
    bottom: theme.spacing(1),
    right: theme.spacing(1),
  },
});
