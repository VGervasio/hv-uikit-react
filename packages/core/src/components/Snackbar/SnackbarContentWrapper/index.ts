export { default as snackbarContentClasses } from "./snackbarContentWrapperClasses";
export * from "./snackbarContentWrapperClasses";
export { default as HvSnackbarContent } from "./SnackbarContentWrapper";
export * from "./SnackbarContentWrapper";
