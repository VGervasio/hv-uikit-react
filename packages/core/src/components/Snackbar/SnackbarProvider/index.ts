export { default as HvSnackbarProvider } from "./SnackbarProvider";
export * from "./SnackbarProvider";
