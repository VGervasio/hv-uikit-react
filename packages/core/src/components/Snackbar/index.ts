export { default as snackbarClasses } from "./snackbarClasses";
export * from "./snackbarClasses";
export * from "./Snackbar";
export * from "./SnackbarProvider";
export * from "./SnackbarContentWrapper";
