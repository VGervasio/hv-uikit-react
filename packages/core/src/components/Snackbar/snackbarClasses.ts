import { getClasses } from "utils";

export type HvSnackbarClasses = {
  /** Styles applied to the component when define as top right.  */
  anchorOriginTopRight?: string;
  /** Styles applied to the component when define as top left.  */
  anchorOriginTopLeft?: string;
  /** Styles applied to the component when define as top center.  */
  anchorOriginTopCenter?: string;
  /** Styles applied to the component when define as bottom center.  */
  anchorOriginBottomCenter?: string;
  /** Styles applied to the component when define as bottom left.  */
  anchorOriginBottomLeft?: string;
  /** Styles applied to the component when define as bottom right.  */
  anchorOriginBottomRight?: string;
};

const classKeys: string[] = [
  "anchorOriginTopRight",
  "anchorOriginTopLeft",
  "anchorOriginTopCenter",
  "anchorOriginBottomCenter",
  "anchorOriginBottomLeft",
  "anchorOriginBottomRight",
];

export const snackbarClasses = getClasses<HvSnackbarClasses>(
  classKeys,
  "HvSnackbar"
);
