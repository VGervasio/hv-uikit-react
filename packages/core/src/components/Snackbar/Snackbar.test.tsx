import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { HvSnackbar } from "./Snackbar";

describe("Snackbar", () => {
  it("should be defined", () => {
    const { container } = render(<HvSnackbar />);
    expect(container).toBeDefined();
  });

  it("should render correctly", () => {
    const { container } = render(<HvSnackbar />);
    expect(container).toMatchSnapshot();
  });

  it("should render the snackbar text correctly", () => {
    const { getByText } = render(
      <HvSnackbar open showIcon variant="success" label="My snackbar" />
    );
    expect(getByText("My snackbar")).toBeInTheDocument();
  });
});
