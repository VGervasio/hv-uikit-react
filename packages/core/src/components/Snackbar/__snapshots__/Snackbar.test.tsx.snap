// Vitest Snapshot v1

exports[`Snackbar > should render correctly 1`] = `<div />`;
